src/components/sign-in.js

```diff
@@ -1,59 +1,183 @@
+import {Alert} from "@material-ui/lab";
+
+const useStyles = makeStyles((theme) => ({
+    paper: {
+        marginTop: theme.spacing(8),
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+    },
+    avatar: {
+        margin: theme.spacing(1),
+        backgroundColor: theme.palette.secondary.main,
+    },
+    form: {
+        width: '100%',
+        marginTop: theme.spacing(3),
+    },
+    submit: {
+        margin: theme.spacing(3, 0, 2),
+    },
+}));
+
+function Copyright() {
+    return (
+        <Typography variant="body2" color="textSecondary" align="center">
+            {'Copyright © '}
+            <Link color="inherit" href="https://github.com/CMSC-495-Group-One">
+                {/* Your Website */}
+                Git Repo
+            </Link>{' '}
+            {new Date().getFullYear()}
+            {'.'}
+        </Typography>
+    );
+}
+
+export default function SignInForm() {
+    const classes = useStyles();
+        const formMethods = useForm();
+        const { signIn } = useAuth();
+        const navigate = useNavigate();
+
+        const {
+            control,
+            register,
+            handleSubmit,
+            formState: { errors },
+        } = formMethods;
+
+        const [isLoading, setIsLoading] = useState(false);
+        const [loginError, setLoginError] = useState(null);
+
+
+        const onSubmit = async (data) => {
+            setIsLoading(true);
+            setLoginError(null);
+            try {
+                await signIn(data);
+                setIsLoading(false);
+                navigate("/tasks");
+            } catch (error) {
+                setIsLoading(false);
+                // handle error, e.g. show a message to the user
+                setLoginError("Invalid username or password.");
+            }
+        };
+
+    // Error Message Div:
+    const errorDiv =loginError
+        ? <Alert severity="error">{loginError}</Alert>
+        : '';
+
+    const [showPassword, setShowPassword] = useState(false);
+
+    const handleClickShowPassword = () => setShowPassword((show) => !show);
+
+    const handleMouseDownPassword = (event) => {
+        event.preventDefault();
+    };
+
+
+    return (
+        <Container component="main" maxWidth="xs">
+            <CssBaseline/>
+            <div className={classes.paper}>
+                <Avatar className={classes.avatar}>
+                    <LockOutlinedIcon/>
+                </Avatar>
+                <Typography component="h1" variant="h5">
+                    Sign In
+                </Typography>
+                <form className={classes.form} onSubmit={handleSubmit(onSubmit)}>
+                    <Grid container spacing={2}>
+                        <Grid item xs={12}>{errorDiv}</Grid>
+                        <Grid item xs={12}>
+                            <InputField
+                                {...register("username")}
```

Comment:
What is the point of using register here if our InputField is already using a Controller?

Reply:
I'll remove them. At some point I was debugging and added these from the old sign in form just to see how it behave